File: storagemanager/src/main/java/com/example/alexeyglushkov/cachemanager/StorageEntry.java
package com.example.alexeyglushkov.cachemanager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Created by alexeyglushkov on 04.10.15.
 */
public interface StorageEntry {
    @Nullable Object getObject() throws Exception;

    void delete() throws Exception;

    @Nullable StorageMetadata getMetadata();
}

File: storagemanager/src/main/java/com/example/alexeyglushkov/cachemanager/StorageEntry.kt
package com.example.alexeyglushkov.cachemanager

/**
 * Created by alexeyglushkov on 04.10.15.
 */
interface StorageEntry {
    fun getObject(): Any?

    @Throws(Exception::class)
    fun delete()

    val metadata: StorageMetadata?
}